### DISCUSSION TITLE
create experimental branch

### DISCUSSION THREAD

**Discussion by trvswgnr** [crablang/crab#105] [Ideas]
just need to create the experimental branch

it will start from the current state of our master branch, which is currently synced with upstream project but afterwards will not receive updates unless they can be incorporated without too many conflicts.

**Comment by pinkforest** [comment]
Does it need to be a branch ? I think branch would get messy here.

I think it should have clear separation that goes against the goal of keeping things compatible.

I also would think a new repo that acts as monorepo with different flavors and would hold .diff's instead of the whole development tree on which there would be tools to combine the upstream and diff's instead when needed - e.g. CI cache and release pipeline.

Moving to discussion as it's experimental.
